test/separate_compilation/jturner/use_array.chpl

```diff
@@ -1,4 +1,4 @@
-proc dothis(r:[1..2] real); 
+proc dothis(r:[1..2] real): real;
```

Comment:
We were getting an earlier complaint about not being able to figure out the return type that's unrelated to the problem this version was trying to expose (separate compilation involving array types gives its own special message).  I have a bug future for that problem elsewhere, so this didn't need to test it